### PR TITLE
Add token as mandatory to get product xml feed (37103)

### PR DESCRIPTION
**Pull request by bogdan202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | get product feed under token only
| Type?         | bug fix
| BC breaks?    | yes, url without token return now 401
| Deprecations? | no
| Fixed ticket? | Fixes 37103
| How to test?  | visit the url of product feed with and without token

### CODE REVIEW

File: controllers/front/product.php [hunk 1]
```diff
@@ -28,36 +28,42 @@ class ShoppingfeedProductModuleFrontController extends \ModuleFrontController
...
+        parent::init();
+
+        $this->sfToken = (new ShoppingfeedToken())->findByToken(Tools::getValue('token', ''));
+    }
 
-        if ($token === false) {
-            exit();
+    public function viewAccess()
+    {
+        if (empty($this->sfToken)) {
```
Comment (clotairer) [inline controllers/front/product.php:40]: perhaps better in checkAccess() method ?

**Review by clotairer — CHANGES_REQUESTED** [review]

File: controllers/front/product.php (hunk 1 above)
Comment (bogdan202) [reply controllers/front/product.php:40]: Yes, checkAccess() is better. I have changed it

File: controllers/front/product.php [hunk 2]
```diff
@@ -28,36 +28,42 @@ class ShoppingfeedProductModuleFrontController extends \ModuleFrontController
...
+
+        $this->sfToken = (new ShoppingfeedToken())->findByToken(Tools::getValue('token', ''));
+    }
 
-        if ($token === false) {
-            exit();
+    public function checkAccess()
+    {
+        if (empty($this->sfToken)) {
+            header('HTTP/1.1 401 Unauthorized');
```
Comment (clotairer) [inline controllers/front/product.php:41]: Could you please use https://www.php.net/manual/en/function.http-response-code.php instead ? It’s compatible with http 1.1 or http 2.0 according to the merchand server

**Review by clotairer — CHANGES_REQUESTED** [review]

File: controllers/front/product.php (hunk 2 above)
Comment (bogdan202) [reply controllers/front/product.php]: Done

**Review by clotairer — APPROVED** [review]

**Pull request merged**